diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/smoke/DiktatSaveSmokeTest.kt

```diff
@@ -0,0 +1,204 @@
+package org.cqfn.diktat.ruleset.smoke
+
+import org.cqfn.diktat.common.config.rules.DIKTAT_COMMON
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.RulesConfigReader
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.rules.DiktatRuleSetProvider
+import org.cqfn.diktat.util.FixTestBase
+
+import com.charleskorn.kaml.Yaml
+import com.charleskorn.kaml.YamlConfiguration
+import com.pinterest.ktlint.core.LintError
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+
+import java.time.LocalDate
+
+import kotlin.io.path.createTempFile
+import kotlinx.serialization.builtins.ListSerializer
+
+class DiktatSaveSmokeTest : FixTestBase("test/smoke/src/main/kotlin",
```

Comment:
Maybe, create a common base class with tests declarations and two implementations with different execution methods? Now we have duplication of all test methods, and it'll be hard to maintain in the future

Reply:
Added base class

Reply:
It still has quite a lot of duplication. IIUC, the only difference is that in one set of tests you are using `runDiktatSmokeTest` method, and in the other `runDiktatSmokeTestWithSave`. So I would expect something like base class with test methods and abstract runner method which is overriden (seems easier, but I'm not sure if Junit will respect `@Test` annotation from base class) or - more complex - a [parameterized tests](https://www.baeldung.com/parameterized-tests-junit-5#6-method) with runner method as a parameter

Reply:
Ok. Changed